crates/matrix-sdk/tests/integration/client.rs

```diff
@@ -583,3 +583,142 @@ async fn cross_signing_status() {
+        })))
+        .mount(&server)
+        .await;
+
+    assert!(
+        room.is_encrypted().await.expect("We should be able to check if the room is encrypted"),
+        "The room should be encrypted"
+    );
+
+    Mock::given(method("GET"))
+        .and(path_regex("/_matrix/client/r0/rooms/!SVkFJHzfwvuaIEawgC:localhost/members"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({ "chunk": []})))
+        .mount(&server)
+        .await;
+
+    let event_content = Arc::new(std::sync::Mutex::new(None));
+
+    let event_content_matcher = {
+        let event_content = event_content.to_owned();
+        move |request: &wiremock::Request| {
+            let path_segments =
+                request.url.path_segments().expect("The URL should be able to be a base");
+
+            let event_type = path_segments
+                .last()
+                .expect("The path should have a event type as the last segment")
+                .to_owned();
+
+            assert_eq!(
+                event_type, "m.room.encrypted",
+                "The event type should be the `m.room.encrypted` event type"
+            );
+
+            let content: RoomEncryptedEventContent = request
+                .body_json()
+                .expect("The uploaded content should be a valid `m.room.encrypted` event content");
+
+            *event_content.lock().unwrap() = Some(content);
+
+            true
+        }
+    };
+
+    Mock::given(method("PUT"))
+        .and(path(
+            "/_matrix/client/r0/rooms/!SVkFJHzfwvuaIEawgC:localhost/send/m.room.encrypted/foobar",
+        ))
+        .and(event_content_matcher)
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+            "event_id": "$foobar"
+        })))
+        .mount(&server)
+        .await;
+
+    room.send_raw("m.room.message", json!({"body": "Hello", "msgtype": "m.text"}))
+        .with_transaction_id("foobar".into())
+        .await
+        .expect("We should be able to send a message to the encrypted room");
+
+    let content = event_content
+        .lock()
+        .unwrap()
+        .take()
+        .expect("We should have intercepted an `m.room.encrypted` event content");
+
+    let event = ruma::serde::Raw::new(&json!({
+        "room_id": room.room_id(),
+        "event_id": "$foobar",
+        "origin_server_ts": 1600000u64,
+        "sender": user_id,
+        "content": content,
+    }))
+    .expect("We should be able to construct a full event from the encrypted event content")
+    .cast();
+
+    let timeline_event = room
+        .decrypt_event(&event)
+        .await
+        .expect("We should be able to decrypt an event that we ourselves have encrypted");
+
+    let event = timeline_event
+        .event
+        .deserialize()
+        .expect("We should be able to deserialize the decrypted event");
+
+    let message_event = assert_matches::assert_matches!(
```

Comment:
cool kids these days use `assert_let!`, i heard :sunglasses:

Reply:
It would be nice if `assert_let!` had a bit more docs, perhaps even an example.

Reply:
Changed in https://github.com/matrix-org/matrix-rust-sdk/pull/2799/commits/ce406d5ae8e1aab8535e3b51244cd86f62d583c6.

Reply:
Will write some tomorrow.

Reply:
Thanks, figured things out via other uses in the SDK.